config/config_auth_details.go

```diff
@@ -0,0 +1,134 @@
+package config
+
+import (
+	"fmt"
+	"strings"
+
+	"golang.org/x/exp/slices"
+)
+
+// AuthDetails contains the details of the authentication configuration.
+type AuthDetails struct {
+	AuthType      string            `json:"auth_type"`
+	Host          string            `json:"host,omitempty"`
+	Configuration AuthConfiguration `json:"configuration"`
+}
+
+func (a *AuthDetails) String() string {
+	s := ""
+	if a.Host != "" {
+		s += fmt.Sprintf(`host=%s`, a.Host)
+	}
+
+	if a.AuthType != "" {
+		s += fmt.Sprintf(", auth_type=%s", a.AuthType)
+	}
+
+	conf := a.Configuration.String()
+	if conf != "" {
+		s += fmt.Sprintf(`
+Configuration:
+%s`, conf)
+	}
+
+	return s
+}
+
+// AuthConfiguration is a map of attribute name to its configuration.
+type AuthConfiguration map[string]*AttrConfig
+
+func (c AuthConfiguration) String() string {
+	keys := make([]string, 0, len(c))
+	for k := range c {
+		keys = append(keys, k)
+	}
+	slices.Sort(keys)
+
+	var conf []string
+	for _, k := range keys {
```

Comment:
Can we store this as a list in the natural order (same as they appear in `ConfigAttributes`)?